shopBuddy/shopBuddy/APIManager.m

```diff
@@ -47,5 +48,28 @@ - (void)getItem:(void(^)(NSDictionary *itemDetails, NSError *error))completion
      }];
     
     
+}
+- (void)getItem:(void(^)(Item *item, NSError *error))completion
+{
+
+    NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile: [[NSBundle mainBundle] pathForResource: @"Keys" ofType: @"plist"]];
+    NSString *key = [dict objectForKey: @"api_key"];
+    NSString *barcode = @"3614272049529";
+    //NSString *barcode_call =@"v3/products?barcode=3614272049529&formatted=y&key=";
+    NSString *path = [NSString stringWithFormat:@"v3/products?barcode=%@&formatted=y&key=%@", barcode, key];
+
+    [manager GET:path parameters:nil headers: nil progress:nil success:^(NSURLSessionTask *task, NSDictionary *responseObject)
+     {
+         // Success
+         NSLog(@"Success: %@", responseObject);
+        Item *item = [[Item alloc] initWithDictionary:responseObject[@"products"][0]];
+        completion(item, nil);
```

Comment:
You could have extracted parsing logic to a private method or helper class to make it more testable.